Automattic/wp-calypso
# Limit Global Styles: Add premium badge to style selection during onboarding

Fixes https://github.com/Automattic/dotcom-forge/issues/896

#### Proposed Changes

Adds the premium badge to the style selection during onboarding.

<img width="200" alt="Screen Shot 2022-10-14 at 16 52 50" src="https://user-images.githubusercontent.com/1233880/195878228-e85b2dd0-fb76-49d7-af34-8cb0da832d3e.png">

To achieve this, I decided to slightly refactor the premium badge component introduced by https://github.com/Automattic/wp-calypso/pull/68841 to better support this use case: 

* The component is now in the generic `components` folder.
* We can now set different tooltip messages.

#### Testing Instructions

- Use the Calypso live link below
- Go to `/start` and create a new free site
- Stop at the "Pick a design" step
- Select a theme with multiple style variations
- In the sidebar, make sure that all the variations except the default one contain a premium badge
- Make sure the premium badge displays a tooltip when hovering it
- Go to `/start` again and create a new paid site now
- Stop at the "Pick a design" step
- Select a theme with multiple style variations
- Make sure the premium badge doesn't show up in any variation
